CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc

```diff
@@ -643,80 +645,184 @@ static function array<name> GetSitrepsForAssaultMission (XComGameState_MissionSi
+				!BonusTemplate.bSitRep // We only consider sitrep bonuses here
+			)
+			{
+				continue;
+			}
 
-		// All good, use the sitrep
-		// We add directly instead of returning the array so that the MeetsRequirements call later accounts for this sitrep
-		MissionState.GeneratedMission.SitReps.AddItem(SitRepTemplate.DataName);
+			SitRepTemplate = SitRepManager.FindSitRepTemplate(BonusTemplate.MetatdataName);
+			if (SitRepTemplate == none || !SitRepTemplate.MeetsRequirements(MissionState))
+			{
+				continue;
+			}
 
-		if (!BonusTemplate.DoNotMarkUsed)
-		{
-			CardManager.MarkCardUsed('OverInfiltrationBonuses', Card);
-		}
+			// All good, use the sitrep
+			MissionState.GeneratedMission.SitReps.AddItem(SitRepTemplate.DataName);
 
-		// We are done
-		break;
-	}
+			if (!BonusTemplate.DoNotMarkUsed)
+			{
+				CardManager.MarkCardUsed('OverInfiltrationBonuses', Card);
+			}
 
-	// Select a negative sitrep. Do this after positive sitreps, since there are more risks and some are not compatible
-	CardManager.GetAllCardsInDeck('FlatRisks', CardLabels);
-	foreach CardLabels(Card)
-	{
-		i = default.FlatRiskSitReps.Find('FlatRiskName', name(Card));
+			// We are done
+			break;
+		}
 
-		if (i != INDEX_NONE)
+		// Select a negative sitrep. Do this after positive sitreps, since there are more risks and some are not compatible
+		CardManager.GetAllCardsInDeck('FlatRisks', CardLabels);
+		foreach CardLabels(Card)
 		{
-			SitRepTemplate = SitRepManager.FindSitRepTemplate(default.FlatRiskSitReps[i].SitRepName);
+			i = default.FlatRiskSitReps.Find('FlatRiskName', name(Card));
 
-			if (SitRepTemplate != none && SitRepTemplate.MeetsRequirements(MissionState))
+			if (i != INDEX_NONE)
 			{
-				MissionState.GeneratedMission.SitReps.AddItem(SitRepTemplate.DataName);
-				break;
+				SitRepTemplate = SitRepManager.FindSitRepTemplate(default.FlatRiskSitReps[i].SitRepName);
+
+				if (SitRepTemplate != none && SitRepTemplate.MeetsRequirements(MissionState))
+				{
+					MissionState.GeneratedMission.SitReps.AddItem(SitRepTemplate.DataName);
+					break;
+				}
 			}
 		}
 	}
 
+	// Select enviromental sitreps
+	// Do this after since this pool is most likely bigger and as such more likely to "survive" exclusions
+	SelectEnviromentalSitreps(MissionState);
+
 	return EmptyArray;
 
 	// Prevent compiler warning
 	EmptyArray.Length = 0;
 }
 
+static function array<name> GetAllEnviromentalSitreps ()
+{
+	local X2StrategyElementTemplateManager StrategyTemplateManager;
+	local X2SitRepTemplateManager SitRepTemplateManager;
+	local X2OverInfiltrationBonusTemplate BonusTemplate;
+	local ActionFlatRiskSitRep FlatRiskSitRep;
+	local X2DataTemplate DataTemplate;
+	local array<name> SitReps;
+
+	StrategyTemplateManager = class'X2StrategyElementTemplateManager'.static.GetStrategyElementTemplateManager();
+	SitRepTemplateManager = class'X2SitRepTemplateManager'.static.GetSitRepTemplateManager();
+
+	// Get all sitreps
+	// TODO: filter SitRep.bExcludeFromStrategy
```

Comment:
Has this TODO been completed yet? If so, remove comment, if not, it seems like an important thing to do.